Tracing update (#2318)
SumoLogic/sumologic-documentation: docs/apm/traces/view-and-investigate-traces.md

The results are not automatically updated. If you want to refresh traces, click the refresh button on the top right corner of the page.<br/>![Refresh.png](/img/traces/Refresh.png)

### Trace Query Visualizations

> The Trace Query Visualizations allow you to visualize the breakdown of where time was spent in traces for various services in your application, complexity of your traces as well as errors. By aggregating traces, you can gain insights into anomalies and unexpected behaviors, leading to faster Time-to-Resolution (TTR) and a higher Return on Investment (ROI). It offers six charts to help you analyze and trace data efficiently. These include:

Reviewer: ```suggestion
Trace Query Visualizations allow you to visualize the breakdown of where time was spent in traces for various services in your application, complexity of your traces as well as errors. By aggregating traces, you can gain insights into anomalies and unexpected behaviors, leading to faster Time-to-Resolution (TTR) and a higher Return on Investment (ROI). It offers six charts to help you analyze and trace data efficiently. These include:
```

Author: @kimsauce Thanks for the suggestion. I believe the usage of "the" here is not incorrect, so I am keeping it as is.

The results are not automatically updated. If you want to refresh traces, click the refresh button on the top right corner of the page.<br/>![Refresh.png](/img/traces/Refresh.png)

### Trace Query Visualizations

> The Trace Query Visualizations allow you to visualize the breakdown of where time was spent in traces for various services in your application. By aggregating traces, you can gain insights into anomalies and unexpected behaviors, leading to faster Time-to-Resolution (TTR) and a higher Return on Investment (ROI). It offers six charts to help you analyze and trace data efficiently. These include:

Reviewer 2: ```suggestion
The Trace Query Visualizations allow you to visualize the breakdown of where time was spent in traces for various services in your application, complexity of your traces as well as errors. By aggregating traces, you can gain insights into anomalies and unexpected behaviors, leading to faster Time-to-Resolution (TTR) and a higher Return on Investment (ROI). It offers six charts to help you analyze and trace data efficiently. These include:
```

> * Trace duration

Reviewer 2: ```suggestion
* Trace duration as timeseries
```

Author: @pbrzoska-sumo I initially thought about specifying the chart type individually. But thank you for the suggestions. I've implemented them.

> * Error count

Reviewer 2: ```suggestion
* Error count as timeseries
```

> * Error count

Reviewer 2: ```suggestion
* Error count as histogram
```

> * Span count

Reviewer 2: ```suggestion
* Span count as timeseries
```

* Span count
* Trace duration as histogram

> 

Reviewer 2: ```suggestion
* **Timeseries**. It shows the average trace duration for each time bucket. It allows you to switch between linear and logarithmic scales to better visualize the data.
* **Histogram**. It shows the distribution of data in different time periods  such as the number of traces with a certain range of spans or errors. You can switch between linear and logarithmic scales to better visualize the data.
```

<img src={useBaseUrl('/img/traces/breakdown2.png')} alt="breakdown2.png" width="950px" height="500" />

> For each chart type, you can choose to view it as either a **timeseries** or **histogram** chart. Here is some more detailed information about each chart type:

Reviewer 2: ```suggestion
Here is some more detailed information about each chart type:
```

> * **Trace duration**. It allows you to quickly identify services that are causing spikes or slowdowns in trace duration. It helps you understand the average trace duration for every time bucket, as well as the amount of time each service contributed to the end-to-end duration.

Reviewer 2: ```suggestion
* **Trace duration**. 
Timeseries: It allows you to quickly identify services that are causing spikes or slowdowns in trace duration. It helps you understand the average trace duration for every time bucket, as well as the amount of time each service contributed to the end-to-end duration.
Histogram: helps you understand the distribution of trace durations. For example you can notice groups of durations responsible for longer traces during slowdown periods.
```

Author: @pbrzoska-sumo I see you have removed the Trace Duration description and added it in Timeseries.

Author: @pbrzoska-sumo : I see you changed the Trace length description to Timeseries.Do we want to leave the Trace duration field blank? Let me know if the below description can be added.
Trace duration: The amount of time each service contributed to the end-to-end duration.

> * **Error count**. It helps you visualize the average number of errors per trace for certain time ranges. You can drill down to specific time ranges and identify the distribution of errors.

Reviewer 2: ```suggestion
* **Error count**. 
Timeseries: It helps you visualise the average number of errors per trace for selected time range.
Histogram: Provides an insight in how many traces were found with certain number of errors. For example if a backend fails in a specific way, it can always generate similar number of errors for traces from that period.
```

> * **Error count**. It helps you visualize the average number of errors per trace for certain time ranges. You can drill down to specific time ranges and identify the distribution of errors.

Reviewer 2: ```suggestion
* **Error count**. 
Timeseries: It helps you visualise the average number of errors per trace for selected time range.
Histogram: Provides an insight in how many traces were found with certain number of errors. For example if a backend fails in a specific way, it can always generate similar number of errors for traces from that period.
```

* **Span count**. It helps you visualize how many spans you have per trace. You can drill down to specific time ranges and identify the distribution of spans.

> These charts provide valuable insights into various aspects of your tracing data performance. It helps you quickly identify any unusual spikes in errors that may indicate an issue in the system.

Reviewer 2: ```suggestion
These charts provide valuable insights into various aspects of your tracing data performance. It helps you quickly identify any unusual spikes or distribution anomalies that may indicate an issue in the system.
```

> You can further switch between the charts from the dropdown menu, and view them as **timeseries** or **histograms**. **Trace duration** is set as the default **timeseries** chart, but you can drill down further and view additional charts such as the histogram to get a more detailed understanding of your tracing data.

Reviewer 2: ```suggestion
You can point-and-drag on chart or click on any column to further narrow down your results. Clicking on a column in timeseries drills-down to its timestamp and service (if present). Clicking on a column in histogram chart narrows down selection of traces matching the values of clicked column.
Results are displayed in trace list below.
```

> * **Timeseries**. It shows the average trace duration for each time bucket. It allows you to switch between linear and logarithmic scales to better visualize the data.

Reviewer 2: ```suggestion
```

> * **Timeseries**. It shows the average trace duration for each time bucket. It allows you to switch between linear and logarithmic scales to better visualize the data.

Reviewer 2: ```suggestion
```

> * **Histogram**. It shows the distribution of data in different time periods  such as the number of traces with a certain range of spans or errors. You can switch between linear and logarithmic scales to better visualize the data.<br/><img src={useBaseUrl('img/traces/charttype.png')} alt="charttype.png" width="950px" />

Reviewer 2: ```suggestion
<br/><img src={useBaseUrl('img/traces/charttype.png')} alt="charttype.png" width="950px" />
```

> To get the best results from this chart, it is recommended to filter the traces to represent similar traces of the same transaction, such as login. Running this chart for different transaction types will not provide the insights users need. Similarly, running the chart for all data without any filters is disabled.

Reviewer 2: ```suggestion
To get the best results from this chart, it is recommended to filter data to represent similar traces of the same transaction, such as login. Running this chart for different transaction types will not provide the insights users need. Similarly, running the chart for all data without any filters is disabled.
```

> * The height of the bar represents the average trace duration for each time bucket.

Reviewer 2: ```suggestion
```

> * Each segment represents a Critical Path Contribution of each service from each trace. Services not present in certain traces do not contribute to the value.

Reviewer 2: ```suggestion
```

> * Click on any of the color segments to focus on this service and drill down to selected timeframe.

Reviewer 2: ```suggestion
```

> * Click and drag on the chart to zoom in.

Reviewer 2: ```suggestion
```

> * Click and drag on the chart to zoom in.

Reviewer 2: ```suggestion
```

The results are not automatically updated. If you want to refresh traces, click the refresh button on the top right corner of the page.<br/>![Refresh.png](/img/traces/Refresh.png)

### Trace Query Visualizations

> The Trace Query Visualizations allow you to visualize the breakdown of where time was spent in traces for various services in your application, the complexity of your traces as well as errors. By aggregating traces, you can gain insights into anomalies and unexpected behaviors, leading to faster Time-to-Resolution (TTR) and a higher Return on Investment (ROI). There are three types of metrics that Trace Query Visualizations offer:

Reviewer: ```suggestion
Trace Query Visualizations allow you to visualize the breakdown of where time was spent in traces for various services in your application, the complexity of your traces as well as errors. By aggregating traces, you can gain insights into anomalies and unexpected behaviors, leading to faster Time-to-Resolution (TTR) and a higher Return on Investment (ROI). There are three types of metrics that Trace Query Visualizations offer:
```

These metrics can be viewed as either a **timeseries** chart or a **histogram** chart.
* **Timeseries**. It shows the average trace duration for each time bucket.
* **Histogram**. It shows the distribution of traces (their count) in different buckets such as duration, number of spans, or errors.

For both charts, you can switch between linear and logarithmic scales to better visualize the data. Note that logarithmic charts will not show any visible bars for buckets with only one trace.

#### Understanding each chart type in detail

* **Trace duration**.
  * **Timeseries**. It allows you to quickly identify services that are causing spikes or slowdowns in trace duration. It helps you understand the average trace duration for every time bucket, as well as the amount of time each service contributed to the end-to-end duration.
    - The height of the bar represents the average trace duration for each time bucket.
    - Each segment represents a Critical Path Contribution of each service from each trace. Services not present in certain traces do not contribute to the value.

>   * **Histogram**. It helps you understand the distribution of trace durations. For example you can notice groups of durations responsible for longer traces during slowdown periods.

Reviewer: ```suggestion
  * **Histogram**. It helps you understand the distribution of trace durations. For example, you can notice groups of durations responsible for longer traces during slowdown periods.
```

 <img src={useBaseUrl('/img/traces/trace-duration-histogram.png')} alt="trace-duration-histogram.png" width="850px" height="700" />

>  Similarly, you can choose timeseries chart type from the dropdown and view the chart.

Reviewer: ```suggestion
 Similarly, you can choose**Timeseries** chart type from the dropdown and view the chart.
```

Author: **timeseries** should be in lowercase to match the UI.

* **Error count**.
  * **Timeseries**. It helps you visualise the average number of errors per trace for selected time range.

>   * **Histogram**. It provides an insight into how many traces were found with a certain number of errors. For example, if a backend fails in a specific way, it can always generate a similar number of errors for traces from that period.

Reviewer: ```suggestion
  * **Histogram**. It provides insight into how many traces were found with a certain number of errors. For example, if a backend fails in a specific way, it can always generate a similar number of errors for traces from that period.
```
